wolfcrypt/src/asn.c

```diff
@@ -17486,41 +17492,31 @@ int CompareOcspReqResp(OcspRequest* req, OcspResponse* resp)
         }
     }
 
-    cmp = XMEMCMP(req->issuerHash, resp->issuerHash, KEYID_SIZE);
-    if (cmp != 0) {
-        WOLFSSL_MSG("\tissuerHash mismatch");
-        return cmp;
-    }
-
-    cmp = XMEMCMP(req->issuerKeyHash, resp->issuerKeyHash, KEYID_SIZE);
-    if (cmp != 0) {
-        WOLFSSL_MSG("\tissuerKeyHash mismatch");
-        return cmp;
-    }
-
     /* match based on found status and return */
-    for (status = resp->status; status; status = next) {
-        cmp = req->serialSz - status->serialSz;
+    for (single = resp->single; single; single = next) {
+        cmp = req->serialSz - single->status->serialSz;
         if (cmp == 0) {
-            cmp = XMEMCMP(req->serial, status->serial, req->serialSz);
+            cmp = XMEMCMP(req->serial, single->status->serial, req->serialSz)
```

Comment:
What if the results from memcmp are 0, 1, and -2?  Don't we end with a false 0?

Reply:
Yeah, this is wrong. John pointed out the same error with this in a different instance.